tests/limits/test_vif_limit.py

```diff
@@ -0,0 +1,85 @@
+            vm.shutdown(verify=True)
+
+        network_uuid = vm.vifs()[0].param_get('network-uuid')
+        existing_vifs = len(vm.vifs())
+
+        logging.info('Give some more vCPUs (default is 1)')
+        vm.param_set('VCPUs-max', vcpus)
+        vm.param_set('VCPUs-at-startup', vcpus)
+
+        logging.info('Create VIFs before starting the VM')
+        for i in range(existing_vifs, vif_limit):
+            vm.create_vif(i, network_uuid=network_uuid)
+
+        vm.start()
+        vm.wait_for_os_booted()
+
+        logging.info('Verify the interfaces exist in the guest')
+        for i in range(0, vif_limit):
+            if vm.ssh_with_result([f'test -d /sys/class/net/{interface_name}{i}']).returncode != 0:
+                guest_error = vm.ssh_with_result(['dmesg | grep -B1 -A3 xen_netfront']).stdout
+                logging.error("dmesg:\n%s", guest_error)
+                assert False, "The interface does not exist in the guest, check dmesg output for errors"
+
+        logging.info('Configure interfaces')
+        config = '\n'.join([f'iface {interface_name}{i} inet dhcp\n'
+                            f'auto {interface_name}{i}'
+                            for i in range(1, vif_limit)])
+        vm.ssh([f'echo "{config}" >> /etc/network/interfaces'])
+
+        logging.info('Install iperf3 on VM and host')
+        if vm.ssh_with_result(['apt install iperf3 --assume-yes']).returncode != 0:
+            assert False, "Failed to install iperf3 on the VM"
+        host.yum_install(['iperf3'])
+
+        logging.info('Reconfigure VM networking')
+        if vm.ssh_with_result(['systemctl restart networking']).returncode != 0:
+            assert False, "Failed to configure networking"
+
+        # Test iperf on all interfaces in parallel
+        # Clean up on exceptions
+        try:
+            logging.info('Create separate iperf servers on the host')
+            host_script = [f'iperf3 -s -p {5100+i} &'
+                           for i in range(0, vif_limit)]
+            host_script = '\n'.join(host_script)
+            host.ssh([f'echo "{host_script}" > host_script'])
+            host.ssh(['nohup bash -c "bash host_script" < /dev/null &>/dev/null &'],
+                     background=True)
+
+            logging.info('Start multiple iperfs on separate interfaces on the VM')
+            vm_script = [f'iperf3 --no-delay -c {host.hostname_or_ip} '
+                         f'-p {5100+i} --bind-dev {interface_name}{i} '
+                         f'--interval 0 --parallel 1 --time 30 &'
+                         for i in range(0, vif_limit)]
+            vm_script = '\n'.join(vm_script)
+            vm.ssh([f'echo "{vm_script}" > vm_script'])
+            stdout = vm.ssh(['bash vm_script'])
+
+            # TODO: log this into some performance time series DB
+            logging.info(stdout)
+        finally:
+            vm.ssh(['kill -9 `pgrep iperf3` || true'])
```

Comment:
`killall -9 iperf3` would not work?

Also a not about the `-9` would be useful - is it necessary for the case the VIF does not work as intended?

Reply:
there's no `killall` on the debian vm

Reply:
would `pkill` be an option?